### ISSUE TITLE
Settings `WATCH_NAMESPACE` to multiple namespaces does not work

### ISSUE THREAD

**Issue by davelosert** [jaegertracing/jaeger-operator#1069]
When setting the `WATCH_NAMESPACE` flag to a comma-separated list as described in the [docs](https://www.jaegertracing.io/docs/1.18/operator/#install-modes), the list actually gets interpreted as one single namespace.

For example setting this:
```
WATCH_NAMESPACE=namespace1,namespace2
```

will result in this error:
```
E0525 08:26:15.626046       1 reflector.go:123] pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/cache/reflector.go:96: Failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:jaeger-operator:jaeger-operator" cannot list resource "deployments" in API group "apps" in the namespace "namespace1,namespace2"
```

This is probably due to a bug in the OperatorSDK (https://github.com/operator-framework/operator-sdk/pull/2522) that got fixed with Version `0.16.0`

However, if I interpret it correctly (I am not a Go-Pro, pun intended), this repository still uses version `0.15.1`.

Would be great if either the version could be increased or the docs could be updated.
I can't do the former (again: not a go-pro 😉 ) but if you say an upgrade of the operator will not be doable, I'll happily create a PR in the [documentation-repository](https://github.com/jaegertracing/documentation) to fix the latter.

**Comment by pavolloffay** [comment]
@davelosert thanks for reporting. 

regarding the docs - is there any workaround or do you want to remove the section that describes how to configure list of namespaces?

In the long run we definitely want to upgrade the operator SDK.

**Comment by davelosert** [comment]
@pavolloffay : Thanks for the quick answer!

Unfortunately, I do not have a workaround.

For the docs: I'd suggest to change
> set the WATCH_NAMESPACE to the comma-separated list of namespaces that the Jaeger Operator should watch

to the singular version:
> set the WATCH_NAMESPACE to the namespace that the Jaeger Operator should watch

And maybe also add a warning box below which explains that setting multiple namespaces is currently not possible with a link to this issue.

**Comment by syck40** [comment]
Hi, I just hit this issue as well.  Is it correct that currently there is no way to do the following?

Without clusterrole binding, I am unable to have 1 clusterwide Operator that will create Jaeger backend in foo namespace and bar namespace as well as inject Jaeger sidecars in those namespaces due to the inability to set WATCH_NAMESPACE = "foo, bar"?

Thanks.

**Comment by jpkrohling** [comment]
You can leave the `WATCH_NAMESPACE` env var empty, and it will act as a cluster-wide operator, requiring the cluster role bindings. Listing the namespaces in `WATCH_NAMESPACE` isn't a widely used pattern, but it should be fixed in the next version of the operator.

**Comment by syck40** [comment]
Ya, I am trying to avoid cluster role, thank you guys.

**Comment by davelosert** [comment]
@jpkrohling : If I see it correctly, the SDK was updated recently - so this should be working now, right? (currently can't try it myself)

**Comment by jpkrohling** [comment]
It *should* indeed work with master at least. Not sure we released a version with this fix already though.

**Comment by lambdanis** [comment]
Can confirm that with the latest release (1.19.0) watching multiple namespaces works as expected for me.

**Comment by jpkrohling** [comment]
Thanks for the confirmation!

**Comment by dangtrinhnt** [comment]
![128524530_1101168410295365_4572481983505400398_n png](https://user-images.githubusercontent.com/2570623/101310134-6ae8ff00-3891-11eb-9a65-a079ee470c69.jpg)
I'm on v1.20.0 and still get this issue.

**Comment by jpkrohling** [comment]
@dangtrinhnt could you please show how you are setting it? Perhaps by sharing your CR.

**Comment by dangtrinhnt** [comment]
I used this operator yaml:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: jaeger-operator
  namespace: jaeger
spec:
  replicas: 1
  selector:
    matchLabels:
      name: jaeger-operator
  template:
    metadata:
      labels:
        name: jaeger-operator
    spec:
      serviceAccountName: jaeger-operator
      containers:
      - name: jaeger-operator
        image: jaegertracing/jaeger-operator:1.20.0
        ports:
        - containerPort: 8383
          name: http-metrics
        - containerPort: 8686
          name: cr-metrics
        args: ["start"]
        imagePullPolicy: Always
        env:
        - name: WATCH_NAMESPACE
          value: md-dev,wpo2-dev,col2-dev
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: OPERATOR_NAME
          value: "jaeger-operator"

```

**Comment by jpkrohling** [comment]
@dangtrinhnt are you sure it's the same issue? Are you getting the same error message as this issue's description? I just tried it here, and I was able to get it working with your CR. The only thing to notice is that you'll need to use the cluster_role.yaml and cluster_role_binding.yaml instead of role.yaml and role_binding.yaml .

**Comment by dangtrinhnt** [comment]
Running it with cluster role will work of-course. I want to limit its permission to specific namespaces only to reduce the risk. I thought using cluster role is just a workaround?

**Comment by jpkrohling** [comment]
> Running it with cluster role will work of-course. I want to limit its permission to specific namespaces only to reduce the risk. I thought using cluster role is just a workaround?

Let me rephrase the question then: have you created a role and role binding for the jaeger-operator service account in the namespaces it needs to act on? For instance, change the role_binding.yaml to specify the jaeger-operator service account's namespace, like this:

```yaml
subjects:
- kind: ServiceAccount
  name: jaeger-operator
  namespace: observability
```

Then create a role/role binding on each namespace you want to watch:

```
kubectl create namespace wpo2-dev
kubectl apply -f deploy/role.yaml -n wpo2-dev
kubectl apply -f deploy/role_binding.yaml -n wpo2-dev
```

**Comment by navjot-puc** [comment]
Steps followed for jaeger installation -
1. `helm install jaeger jaegertracing/jaeger-operator -n jaeger`
2. Created a jaeger instance to watch ingress namespace, sidecar was injected successfully (instances was created in ingress namespace)
3. These are the Nginx logs for jaeger sidecar
![image](https://user-images.githubusercontent.com/84508323/127464314-d0d0b4e3-003b-45ae-8554-bd446c1fcd3d.png)

4. logs for jaeger instance that was created in ingress namespace
![image](https://user-images.githubusercontent.com/84508323/127464490-1a551e42-4e04-462c-b90c-aea0588e5282.png)

5. then there's jaeger operator logs
![image](https://user-images.githubusercontent.com/84508323/127464584-d305ac6b-4b59-4ce0-ab35-71c66298b28e.png)

I've installed the sa,role and rolebinding into the ingress namespace where i want the jaeger to act on, but jaeger-operator logs are showing something else. I can't see anything into the UI as well. 

I can see the sidecar injection into the ingress deployment. I've also enabled the jaeger operator to watch all namespaces.

can anyone help? 

Thanks.

**Comment by jpkrohling** [comment]
@navjot-puc are you sure this is the same problem?

**Comment by navjot-puc** [comment]
@jpkrohling thanks for the response.

I didn't quite get your question? 

Goal:
I'm trying to trace ingress and I've made the relevent changes to the ingress as well and added the annotations to the namespace of ingress and it injected the sidecar automatically but something is wrong according to jaeger operator. I can't wrap my head around this!

Thanks.

**Comment by jpkrohling** [comment]
Are you sure the problem you are reporting is the same that was this ticket is tracking? If not, please open a new issue.

**Comment by navjot-puc** [comment]
ohh, got it.
